### PR TITLE
Rework context propagation and switch to json.RawMessage

### PR DESCRIPTION
**Pull request by FinnStutzenstein**

### CODE REVIEW

File: internal/allowed/topic/create_test.go [hunk 1]
```diff
@@ -53,119 +54,131 @@ func assertCreateIsAllowed(t *testing.T, params *allowed.IsAllowedParams) {
...
-	dp.AddPermissionToGroup(1, "agenda.can_manage")
-	data := definitions.FqfieldData{
-		"meeting_id": "1",
-	}
-	params := &allowed.IsAllowedParams{UserID: 0, Data: data, DataProvider: dp.GetDataprovider()}
-
-	assertCreateIsAllowed(t, params)
+func TestCreate(t *testing.T) {
+	t.Run("UnknownUser", func(t *testing.T) {
+		dp := tests.NewTestDataProvider(context.TODO())
```
Comment (ostcar) [inline internal/allowed/topic/create_test.go:59]: From a technical view, `context.TODO()` and `context.Background()` do exactly the same. The Name `TODO` should only be used, when there is something, the developer should do.

I don't know, if you want to do something else here in the near future. In this case, the context `context.TODO()` is correct. But if you want this code to stay this way, then `context.Background()` would be the right choice.

File: internal/core/permission.go [hunk 2]
```diff
@@ -11,27 +12,28 @@ import (
...
 type PermissionService struct {
-	dataprovider dataprovider.DataProvider
+	externalDataprovider dataprovider.ExternalDataProvider
 }
 
 // NewPermissionService returns a new permission service.
 func NewPermissionService(externalDataprovider dataprovider.ExternalDataProvider) *PermissionService {
-	dp := dataprovider.NewDataProvider(externalDataprovider)
-	return &PermissionService{dp}
+	return &PermissionService{externalDataprovider}
```
Comment (ostcar) [inline internal/core/permission.go:20]: I would give the function attribute `externalDataprovider` a shorter name. The scope of the variable is only two lines. So a name like `edp` is not a problem.

See also: https://github.com/golang/go/wiki/CodeReviewComments#variable-names

File: internal/core/permission.go [hunk 3]
```diff
@@ -11,27 +12,28 @@ import (
...
 }
 
 // IsAllowed tells, if something is allowed.
-func (permissionService PermissionService) IsAllowed(name string, userID int, data definitions.FqfieldData) (bool, map[string]interface{}, error) {
+func (permissionService PermissionService) IsAllowed(ctx context.Context, name string, userID int, data definitions.FqfieldData) (bool, map[string]interface{}, error) {
```
Comment (ostcar) [inline internal/core/permission.go:24]: https://github.com/golang/go/wiki/CodeReviewComments#receiver-names

File: internal/core/permission_test.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ var Queries = map[string]allowed.IsAllowed{
 func TestDispatchNotFound(t *testing.T) {
 	core.Queries = Queries
 	p := core.NewPermissionService(nil)
-	result, addition, err := p.IsAllowed("", 0, nil)
+	result, addition, err := p.IsAllowed(nil, "", 0, nil)
```
Comment (ostcar) [inline internal/core/permission_test.go:21]: You should never use a nil-context. In this place, `context.Background()` is probably the right choice.

In the doc is says:

https://golang.org/pkg/context/
> Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use.

File: internal/http/http.go [hunk 5]
```diff
@@ -24,16 +25,16 @@ func IsAllowed(mux *http.ServeMux, provider IsAlloweder) {
...
+			UserID int                        `json:"user_id"`
+			Data   map[string]json.RawMessage `json:"data"`
 		}
 		if err := json.NewDecoder(r.Body).Decode(&requestData); err != nil {
 			handleError(w, jsonError{"Can not decode request body", err})
 			return
 		}
 
-		allowed, addition, err := provider.IsAllowed(requestData.Name, requestData.UserID, requestData.Data)
+		allowed, addition, err := provider.IsAllowed(context.TODO(), requestData.Name, requestData.UserID, requestData.Data)
```
Comment (ostcar) [inline internal/http/http.go:37]: You can/should use `r.Context()` here

**Comment by FinnStutzenstein** [comment]
Thanks fot the remarks; they are done in #9 and #10

**Pull request merged**